spec/client_spec.rb

```diff
@@ -115,5 +115,125 @@
         client.address.retrieve('adr_123')
       }.to raise_error(EasyPost::Errors::NotFoundError) # should throw error because our lambda returns 404
     end
+
+    describe 'hooks' do
+      after { EasyPost::Hooks.send(:subscribers).clear }
+
+      it 'subscribes to request events' do
+        notifications = []
+        client = described_class.new(api_key: ENV['EASYPOST_TEST_API_KEY'])
+        client.subscribe_request_hook do |request_data|
+          expect(request_data).to be_an(EasyPost::Hooks::RequestContext)
+          expect(request_data.method).to eq(:get)
+          expect(request_data.path).to end_with('/addresses/adr_123')
+          expect(request_data.headers).to be_a(Hash)
```

Comment:
issue: asserting that this is a hash doesn't unfortunately test that these are actual header values. We should probably test for a value we know will actually be in the headers. See other libs for an example.

Reply:
Or at least test that it's not an empty hash, that it has at least one element in it?

Reply:
Added on line 130